Sources/WalletConnect.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  Copyright © 2019 Gnosis Ltd. All rights reserved.
+//
+
+public class WalletConnect {
+
+    let communicator: Communicator
+
+    init() {
+        communicator = Communicator()
+    }
+
+    enum WalletConnectError: Error {
+        case tryingToConnectExistingSessionURL
+        case tryingToDisconnectInactiveSession
+        case missingWalletInfoInSession
+    }
+
+    /// Connect to WalletConnect url
+    /// https://docs.walletconnect.org/tech-spec#requesting-connection
+    ///
+    /// - Parameter url: WalletConnect url
+    /// - Throws: error on trying to connect to existing session url
+    public func connect(to url: WCURL) throws {
+        guard communicator.session(by: url) == nil else {
+            throw WalletConnectError.tryingToConnectExistingSessionURL
+        }
+        listen(on: url)
+    }
+
+    /// Reconnect to the session
+    ///
+    /// - Parameter session: session object with wallet info.
+    /// - Throws: error if wallet info is missing
+    public func reconnect(to session: Session) throws {
+        guard session.walletInfo != nil else {
+            throw WalletConnectError.missingWalletInfoInSession
+        }
+        communicator.addSession(session)
+        listen(on: session.url)
+    }
+
+    /// Disconnect from session.
+    ///
+    /// - Parameter session: Session object
+    /// - Throws: error on trying to disconnect inacative sessoin.
+    public func disconnect(from session: Session) throws {
+        guard communicator.isConnected(by: session.url) else {
+            throw WalletConnectError.tryingToDisconnectInactiveSession
+        }
+        try sendDisconnectSessionRequest(for: session)
+        communicator.addPendingDisconnectSession(session)
+        communicator.disconnect(from: session.url)
+    }
+
+    /// Get all sessions with active connection.
+    ///
+    /// - Returns: sessions list.
+    public func openSessions() -> [Session] {
+        return communicator.openSessions()
+    }
+
+    private func listen(on url: WCURL) {
+        communicator.listen(on: url,
+                            onConnect: onConnect(to:),
+                            onDisconnect: onDisconnect(from:error:),
+                            onTextReceive: onTextReceive(_:from:))
+    }
+
+    /// Confirmation from Transport layer that connection was successfully established.
+    ///
+    /// - Parameter url: WalletConnect url
+    func onConnect(to url: WCURL) {
+        preconditionFailure("Should be implemented in subclasses")
+    }
+
+    //// Confirmation from Transport layer that connection was dropped.
+    ///
+    /// - Parameters:
+    ///   - url: WalletConnect url
+    ///   - error: error that triggered the disconnection
+    private func onDisconnect(from url: WCURL, error: Error?) {
+        print("WC: didDisconnect url: \(url.bridgeURL.absoluteString)")
```

Comment:
debug print in production code, please replace by logging or remove

Reply:
created issue for that